packages/block-editor/src/store/reducer.js

```diff
@@ -1207,6 +1207,22 @@ export const blockListSettings = ( state = {}, action ) => {
 	return state;
 };
 
+/**
+ * Reducer returning whether the navigation mode is enabled or not.
+ *
+ * @param {string} state  Current state.
+ * @param {Object} action Dispatched action.
+ *
+ * @return {string} Updated state.
+ */
+export function isNavigationMode( state = true, action ) {
+	if ( action.type === 'TOGGLE_NAVIGATION_MODE' ) {
+		return action.isNavigationMode;
```

Comment:
With "toggle" being the name, I'd expect: `return ! state`. Don't know if it could be done like that or if it could be renamed. Since the component is checking the state anyway before dispatching any actions, you could go for `return ! state`?

Reply:
I like the `setNavigationMode` proposal.